crates/cli-support/src/js/mod.rs

```diff
@@ -842,7 +847,43 @@ impl<'a> Context<'a> {
         Ok(())
     }
 
-    fn gen_init(&mut self, module_name: &str, needs_manual_start: bool) -> String {
+    fn ts_for_init_fn(has_memory: bool) -> &'static str {
+        macro_rules! initial_ts {
+            () => ("
+                \n\
+                /**\n\
+                * If `module_or_path` is {RequestInfo}, makes a request and\n\
+                * for everything else, calls `WebAssembly.instantiate` directly.\n\
+                *\n\
+                * @param {RequestInfo | BufferSource | WebAssembly.Module} module_or_path\n\
+            ");
+        }
+        if has_memory {
+            concat!(initial_ts!(), "\
+            * @param {WebAssembly.Memory} maybe_memory\n\
+            *\n\
+            * @returns {Promise<WebAssembly.ResultObject | WebAssembly.Instance>}\n\
+            */\n\
+            export function init (\
+                module_or_path: RequestInfo | BufferSource | WebAssembly.Module,\
+                maybe_memory: WebAssembly.Memory\
```

Comment:
This should be marked as optional param (both here and in JSDoc).

Reply:
Although `maybe_memory?` will make linters happier, it is going to be odd to see a parameter that is not being used for anything in some situations.

Reply:
Yes this shouldn't be optional because it's a static property whether it's required or not.

Reply:
Ah this is generated only when memory *has* to be passed from the outside? As in, can I still omit it when memory is defined inside of WASM?

Reply:
Correct yeah

Reply:
I'm going to submit another issue / PR, but just ran into this and appears that it has to be optional after all.